### PR TITLE
added WordPress doc first draft

### PR DESCRIPTION
**Pull request by lucatume**
Here is a first draft for the "For WordPress" page.

### CODE REVIEW

File: for/wordpress.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+If you are using the `WPLoader` module in your tests take care to create a dedicated database for it and not to use the same database the `Db` or `WPDb` modules might use.  
+The use of the modules defined in the WPBrowser package is not tied to this bootstrap though so feel free to set up Codeception in any other way.
+
+## WordPress Unit Tests
+Most references to "WordPress unit tests" imply a reference to [PhpUnit](https://phpunit.de/ "PHPUnit – The PHP Testing Framework") powered tests where, but, a local WordPress instance is loaded in the same scope as the tests before the test methods are executed.  
+So "WordPress" as WordPress is loaded before the tests and "unit tests" as the subject under test is a single class or a group of tightly related functions that can be thought about as a unit.  
+By default WPBrowser creates the `wpunit` suite to keep these tests separate from "real" unit tests stored in the `unit` suite.  
+The `wpunit` suite includes the `WPLoader` module: this takes care of loading, installing and configuring a fresh WordPress installation before each test method runs.
```
Comment (DavertMik) [inline for/wordpress.md]: Ok, I got the point but to me the wording can be improved. Actually that was pretty hard to read with too much unit test and wordpress words in different context.

I think we can avoid this confusion by calling wordpress-unit-tests to be **integration** tests which is actually correct.

File: for/wordpress.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                - acme-plugin/plugin.php
+```
+
+The module is wrapping and augmenting the [WordPress PHPUnit automated testing suite](https://make.wordpress.org/core/handbook/testing/automated-testing/phpunit/) and to generate a test case that uses Codeception and the methods provided by the Core testing suite you can use the `wpcept` command again:
+
+```bash
+wpcept generate:wpunit wpunit "Acme\SomeClass"
+```
+
+The generated test case extends the `Codeception\TestCase\WPTestCase` class and it exposes all the mehtods defined by `Codeception\TestCase\Test` test case and the Core suite `WP_UnitTestCase`.
```
Comment (DavertMik) [inline for/wordpress.md]: `Codeception\TestCase\Test` is now `Codeception\Test\Unit` (in 2.2)
there is still working alias for that class so it should work but better stick to new naming

File: for/wordpress.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+## WordPress Functional Tests
+Functional tests are meant to test complex interactions between a WordPress theme or plugin components, requests handling or a filter triggered chain of events.  
+A WordPress unit test might test the behaviour of a post insertion handling class in isolation where a functional test might verify the behaviour of said class when part of a chain of classes handling a post insertion request.  
+Similarly to WordPress Unit Tests the `functional` suite should include the `WPLoader` module to have WordPress running in the same scope as the test methods and the configuration is the same.
+Since the difference is subtle in infrastructure terms the command to generate a functional test is the same used to generate a WordPress unit test:
+
+```bash
+wpcept generate:wpunit functional "PostInsertionHandling"
+```
```
Comment (DavertMik) [inline for/wordpress.md]: sorry I didn't understand actual difference between functional and wpunit 
If there is no obvious distinction between them maybe they can be used in one suite?

File: for/wordpress.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        WPDb:
+            dsn: 'mysql:host=localhost;dbname=wordpress'
+            user: root
+            password: root
+            dump: tests/_data/wp.sql
+            populate: true
+            cleanup: true
+            url: 'http://wordpress.dev'
+            tablePrefix: wp_
+        WPWebDriver:
```
Comment (DavertMik) [inline for/wordpress.md]: modules WPBrowser and WPWebDriver must not be used together as they share same interface but different backend. I think turning them both on in Codeception 2.2 will throw a conflict exception.

I'd recommend to use `WPBrowser` in functional test suite and `WPWebDriver` in acceptance test suite. The difference would be that functional test functionality and acceptance would test UI. We can even call that suite `ui` or `e2e` for better understanding.

File: for/wordpress.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            adminUsername: admin
+            adminPassword: password
+            adminUrl: /wp-admin
+```
+
+As the `WebDriver` module the `WPWebDriver` module require a Selenium or PhantomJS server to run.
+The `WPDb` module allows for quick interactions with the WordPress database API like quick generation of multiple posts, database value fetching and more:
+
+```php
+$I->haveManyPostsInDatabase(10, ['post_type' => 'custom_post_type']);
```
Comment (DavertMik) [inline for/wordpress.md]: impressive :+1:

**Comment by DavertMik** [comment]
Amazing! Thanks, for writing!

**Comment by lucatume** [comment]
Ok, I've made updates to the doc and, on parallel, to wp-browser to align. Let me know how this looks.

File: for/wordpress.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+To fully use the WordPress specific modules of the WPBrowser suite you need to setup a local WordPress installation; this is in no way different from what's required to locally develop a WordPress theme or plugin.  
+In the examples below I assume that the local WordPress installation root directory is `/var/www/wordpress`, that it responds at the `http://wordpress.dev` URL and that I am developing a plugin called "Acme Plugin".  
+While it's possible to start from a default Codeception configuration the WPBrowser package comes with its own CLI tool (`wpcept`) and its own bootstrap command.  
+Scaffold the tests using the latter
+
+```bash
+wpcept bootstrap
+```
+
+The command creates the `tests` directory and scaffold the `acceptance`, `functional`, `wpunit` and `unit` suites creating a `tests` sub-directory and a configuration file for each.
```
Comment (DavertMik) [inline for/wordpress.md]: but here `wpunit` suite is still created. Should you update wpcept?

File: for/wordpress.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+wpcept bootstrap
+```
+
+The command creates the `tests` directory and scaffold the `acceptance`, `functional`, `wpunit` and `unit` suites creating a `tests` sub-directory and a configuration file for each.  
+Edit the `codeception.yml` file to match your local WordPress setup and point the modules to the local WordPress installation folder, the local WordPress URL and so on.  
+If you are using the `WPLoader` module in your tests take care to create a dedicated database for it and not to use the same database the `Db` or `WPDb` modules might use.  
+The use of the modules defined in the WPBrowser package is not tied to this bootstrap though so feel free to set up Codeception in any other way.
+
+## Integration Tests
+Most references to "WordPress unit tests" imply a reference to [PhpUnit](https://phpunit.de/ "PHPUnit – The PHP Testing Framework") powered tests where, but, a local WordPress instance is loaded in the same scope as the tests before the test methods are executed; by default WPBrowser creates the `integration` suite to store this kind of tests.
```
Comment (DavertMik) [inline for/wordpress.md]: still i don't like wording. Better to say:

> Commonly "WordPress unit tests" are not related to classical unit tests but to integration tests. The difference is that unit tests are supposed to test a class in method in complete isolation, while integration tests check how components work inside WordPress.That's why, to prepare WordPress for testing you should enable `WPLoader` module into `integration.suite.yml`.

File: for/wordpress.md (hunk 6 above)
Comment (lucatume) [inline for/wordpress.md]: I'm pushing modifications to wp-browser `master` branch to match

File: for/wordpress.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+The use of the modules defined in the WPBrowser package is not tied to this bootstrap though so feel free to set up Codeception in any other way.
+
+## Integration Tests
+Commonly "WordPress unit tests" are not related to classical unit tests but to integration tests. The difference is that unit tests are supposed to test a class methods in complete isolation, while integration tests check how components work inside WordPress. That's why, to prepare WordPress for testing, you should enable `WPLoader` module into `integration.suite.yml`.  
+The `WPLoader` module: it takes care of loading, installing and configuring a fresh WordPress installation before each test method runs.  
+To handle the heavy lifting the module requires some information about the local WordPress installation: in the `codeception.yml` file configure it to match your local setup; with reference to the example above the module configuration might be:
+
+```yaml
+modules:
+    enabled:
```
Comment (DavertMik) [inline for/wordpress.md]: I'm not quite sure but inside `codeception.yml` it is better to pass configuration as

```
modules:
    config:
```

Comment (lucatume) [inline for/wordpress.md]: You are right, my mistake.

**Comment by DavertMik** [comment]
Thank you!

**Comment by lucatume** [comment]
Great, thanks to you!

**Pull request merged**